brain_observatory_utilities/utilities/general_utilities.py

```diff
@@ -47,4 +50,459 @@ def validate_value_in_dict_keys(input_key, dictionary, dict_name):
         name of dictionary
     """
     assert input_key in dictionary, "Error: input value is\
-        not in {} keys.".format(dict_name)
\ No newline at end of file
+        not in {} keys.".format(dict_name)
+
+
+def get_time_array(t_start, t_end, sampling_rate=None, step_size=None, include_endpoint=True):  # NOQA E501
+    '''
+    A function to get a time array between two specified timepoints at a defined sampling rate  # NOQA E501
+    Deals with possibility of time range not being evenly divisible by desired sampling rate  # NOQA E501
+    Uses np.linspace instead of np.arange given decimal precision issues with np.arange (see np.arange documentation for details)  # NOQA E501
+
+    Parameters:
+    -----------
+    t_start : float
+        start time for array
+    t_end : float
+        end time for array
+    sampling_rate : float
+        desired sampling of array
+        Note: user must specify either sampling_rate or step_size, not both
+    step_size : float
+        desired step size of array
+        Note: user must specify either sampling_rate or step_size, not both
+    include_endpoint : Boolean
+        Passed to np.linspace to calculate relative time
+        If True, stop is the last sample. Otherwise, it is not included.
+            Default is True
+
+    Returns:
+    --------
+    numpy.array
+        an array of timepoints at the desired sampling rate
+
+    Examples:
+    ---------
+    get a time array exclusive of the endpoint
+    >>> t_array = get_time_array(
+        t_start=-1,
+        t_end=1,
+        step_size=0.5,
+        include_endpoint=False
+    )
+
+    np.array([-1., -0.5,  0.,  0.5])
+
+
+    get a time array inclusive of the endpoint
+    >>> t_array = get_time_array(
+        t_start=-1,
+        t_end=1,
+        step_size=0.5,
+        include_endpoint=False

```

Comment:
shouldn't this be include_endpoint=True?

Reply:
Not sure what Doug’s rationale for this was but if you think it should be True we can change it